academy_project/src/test/kotlin/com/infinum/course/car/checkup/carcheckup/controller/CarCheckUpControllerTest.kt

```diff
@@ -2,28 +2,29 @@ package com.infinum.course.car.checkup.carcheckup.controller
 
 import com.infinum.course.car.checkup.car.entity.Car
 import com.infinum.course.car.checkup.carcheckup.controller.dto.CarCheckUpDTO
-import com.infinum.course.car.checkup.carcheckup.controller.resource.CarCheckUpAnalyticsResourceAssembler
-import com.infinum.course.car.checkup.carcheckup.controller.resource.CarCheckUpResourceAssembler
 import com.infinum.course.car.checkup.carcheckup.entity.CarCheckUp
 import com.infinum.course.car.checkup.carcheckup.service.CarCheckUpService
 import com.infinum.course.car.checkup.cardata.entity.CarData
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import org.mockito.Mockito
+import org.mockserver.springtest.MockServerTest
 import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
-import org.springframework.boot.test.context.TestConfiguration
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
 import org.springframework.boot.test.mock.mockito.MockBean
-import org.springframework.context.annotation.Bean
 import org.springframework.http.MediaType
+import org.springframework.security.oauth2.jwt.JwtDecoder
 import org.springframework.test.web.servlet.MockMvc
 import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers
 import java.time.LocalDate
 import java.time.LocalDateTime
 import java.util.*
 
-@WebMvcTest(CarCheckUpController::class)
+@SpringBootTest
```

Comment:
Hjoj, ovo sam istraživao jučer i zapravo čim smo se nekako maknuli s huddlea sam se sjetio kako se ovo handlea. Problem je što sam na callu gledao materijale iz akademije koji isto uvode `@SpringBootTest` i dignu sve žive dependencyje umjesto samo security, trebat ću to ispravit.

Zapravo ovo možeš vrlo jednostavno optimizirat da ne digneš cijeli kontekst. Sve što zapravo trebaš je dodatno importat security konfiguraciju, a to možeš napravit ovak:
```suggestion
@Import(SecurityConfiguration::class)
@WebMvcTest(CarCheckUpController::class)
```

Reply:
Aaaa niceee, hvala ti, ima tih anotacija stvarno dosta, tko bi ih sve pohvatao sto se moze s kojom haha 😄